src/render/decorations/CHyprGroupBarDecoration.cpp

```diff
@@ -364,11 +387,13 @@ bool CHyprGroupBarDecoration::onBeginWindowDragOnDeco(const Vector2D& pos) {
 }
 
 bool CHyprGroupBarDecoration::onEndWindowDragOnDeco(const Vector2D& pos, PHLWINDOW pDraggedWindow) {
+    static auto PSTACKED = CConfigValue<Hyprlang::INT>("group:groupbar:stacked");
     if (!pDraggedWindow->canBeGroupedInto(m_pWindow.lock()))
         return false;
 
     const float BARRELATIVEX = pos.x - assignedBoxGlobal().x - m_fBarWidth / 2;
-    const int   WINDOWINDEX  = BARRELATIVEX < 0 ? -1 : (BARRELATIVEX) / (m_fBarWidth + BAR_HORIZONTAL_PADDING);
+    const float BARRELATIVEY = pos.y - assignedBoxGlobal().y;
```

Comment:
here you likely need to subtract half a bar height like above

Reply:
I'm not sure it needs to be here, but I do think something needs to be adjusted for the 'half bar' thing. I'll look today